src/redux/slices/flattened-sublayers-slice.tsx

```diff
@@ -32,54 +42,118 @@ const flattenedSublayersSlice = createSlice({
   reducers: {
     setFlattenedSublayers: (
       state: flattenedSublayersState,
-      action: PayloadAction<BuildingSceneSublayerExtended[]>
+      action: PayloadAction<{
+        layers: BuildingSceneSublayerExtended[];
+        side?: ComparisonSideMode;
+      }>
     ) => {
-      state.layers = action.payload;
+      switch (action.payload.side) {
+        case ComparisonSideMode.left:
+          state.left.layers = action.payload.layers;
+          break;
+        case ComparisonSideMode.right:
+          state.right.layers = action.payload.layers;
+          break;
+        default:
+          state.single.layers = action.payload.layers;
+      }
     },
     updateLayerVisibility: (
       state: flattenedSublayersState,
-      action: PayloadAction<{ index: number; visibility: boolean | undefined }>
+      action: PayloadAction<{
+        index: number;
+        visibility: boolean | undefined;
+        side?: ComparisonSideMode;
+      }>
     ) => {
-      if (action.payload.index in state.layers) {
-        state.layers[action.payload.index].visibility =
-          action.payload.visibility;
+      let destLayer: BuildingSceneSublayerExtended | undefined = undefined;
+      switch (action.payload.side) {
+        case ComparisonSideMode.left:
+          if (action.payload.index in state.left.layers) {
+            destLayer = state.left.layers[action.payload.index];
+          }
+          break;
+        case ComparisonSideMode.right:
+          if (action.payload.index in state.right.layers) {
+            destLayer = state.right.layers[action.payload.index];
+          }
+          break;
+        default:
+          destLayer = state.single.layers[action.payload.index];
+      }
+      if (destLayer) {
+        destLayer.visibility = action.payload.visibility;
       }
     },
   },
+
   extraReducers: (builder) => {
     builder
-      .addCase(
-        getFlattenedSublayers.pending,
-        (state: flattenedSublayersState) => {
-          state.layerCounter += 1;
+      .addCase(getFlattenedSublayers.pending, (state, action) => {
```

Comment:
could you add types? meta.arg.side is not clear

Reply:
As per redux's TypeScript usage guide it's already typed - https://redux.js.org/usage/usage-with-typescript#typing-extrareducers